core/server/data/migrations/versions/3.12/001-add-identity-permission.js

```diff
@@ -0,0 +1,54 @@
+const ObjectId = require('bson-objectid');
+const common = require('../../../../lib/common');
+
+module.exports.config = {
+    transaction: true
+};
+
+module.exports.up = async (options) => {
+    const connection = options.transacting;
+
+    const existingIdentityPermission = await connection('permissions').where({
+        action_type: 'read',
+        object_type: 'identity'
+    }).first();
+
+    if (existingIdentityPermission) {
+        common.logging.warn('Adding permission for read:identity');
```

Comment:
```suggestion
        common.logging.warn('Permission read:identity already present');
```

Similar reasoning as below

Reply:
much better!